src/liballoc/vec.rs

```diff
@@ -603,6 +603,92 @@ impl<T> Vec<T> {
         self.buf.try_reserve_exact(self.len, additional)
     }
 
+    /// Allows re-interpreting the type of a Vec to reuse the allocation.
+    /// The vector is emptied and any values contained in it will be dropped.
```

Comment:
```suggestion
    /// The vector is emptied and any values contained in it will be dropped.
    /// As a result, no elements of type `T` are transmuted to `U` and so this operation is safe.
```